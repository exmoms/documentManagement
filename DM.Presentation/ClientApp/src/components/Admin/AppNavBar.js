import AppBar from "@material-ui/core/AppBar";
import IconButton from "@material-ui/core/IconButton";
import Menu from "@material-ui/core/Menu";
import MenuItem from "@material-ui/core/MenuItem";
// Importing Material-UI components ---------------------------
import { makeStyles } from "@material-ui/core/styles";
import Toolbar from "@material-ui/core/Toolbar";
import Typography from "@material-ui/core/Typography";
import AccountCircle from "@material-ui/icons/AccountCircle";
import MoreIcon from "@material-ui/icons/MoreVert";
import HelpOutlineIcon from "@material-ui/icons/HelpOutline";
import React from "react";
import { useTranslation } from "react-i18next";
//--------------------------------------------------
import { Link } from "react-router-dom";
import { useAuth } from "../../context/auth";
import { fetchData } from "../../api/FetchData";
import { postDataToAPI } from "../../api/PostData";

// Add custom style
const useStyles = makeStyles((theme) => ({
  grow: {
    flexGrow: 1,
  },
  menuButton: {
    marginRight: theme.spacing(2),
  },
  title: {
    display: "none",
    [theme.breakpoints.up("sm")]: {
      display: "block",
    },
  },
  inputInput: {
    padding: theme.spacing(1, 1, 1, 7),
    transition: theme.transitions.create("width"),
    width: "100%",
    [theme.breakpoints.up("md")]: {
      width: 200,
    },
  },
  sectionDesktop: {
    display: "none",
    [theme.breakpoints.up("md")]: {
      display: "flex",
    },
  },
  sectionMobile: {
    display: "flex",
    [theme.breakpoints.up("md")]: {
      display: "none",
    },
  },
}));

// Main component function
export default function PrimarySearchAppBar(props) {
  const { t } = useTranslation();
  const classes = useStyles();
  const [anchorEl, setAnchorEl] = React.useState(null);
  const [mobileMoreAnchorEl, setMobileMoreAnchorEl] = React.useState(null);

  const isMenuOpen = Boolean(anchorEl);
  const isMobileMenuOpen = Boolean(mobileMoreAnchorEl);
  const [user_info, setUser_info] = React.useState({});

  React.useEffect(() => {
    fetchData("/api/user/CurrentUserProfile").then((res) => {
      setUser_info(res);
    });
  }, [user_info.userName]);

  const handleProfileMenuOpen = (event) => {
    setAnchorEl(event.currentTarget);
  };

  const handleMobileMenuClose = () => {
    setMobileMoreAnchorEl(null);
  };

  const handleMenuClose = () => {
    setAnchorEl(null);
    handleMobileMenuClose();
  };

  const handleMobileMenuOpen = (event) => {
    setMobileMoreAnchorEl(event.currentTarget);
  };
  /*function setCookie(cname, cvalue, exdays) {
    var d = new Date();
    d.setTime(d.getTime() + exdays * 24 * 60 * 60 * 1000);
    var expires = "expires=" + d.toUTCString();
    document.cookie = cname + "=" + cvalue + ";" + expires + ";path=/";
  }*/

  const { setAuthenticated } = useAuth();
  const logout = (event) => {
    postDataToAPI("/api/user/Logout").then((res) => {
      if (res.status === 200) {
        setAuthenticated(false);
        localStorage.removeItem("isAuthenticated");
      } else {
        let errorObject = {
          open: true,
          error: true,
          errorMessage: [t("logout_error_msg")],
        };
        props.onError(errorObject);
      }
    });
  };

  const menuId = "primary-search-account-menu";
  const renderMenu = (
    <Menu
      anchorEl={anchorEl}
      anchorOrigin={{ vertical: "top", horizontal: "right" }}
      id={menuId}
      keepMounted
      transformOrigin={{ vertical: "top", horizontal: "right" }}
      open={isMenuOpen}
      onClose={handleMenuClose}
    >
      <MenuItem onClick={handleMenuClose} component={Link} to="/profile">
        {t("Profile")}
      </MenuItem>
      {(props.user.role === "Admin" || props.user.role === "SuperAdmin") && (
        <MenuItem onClick={handleMenuClose} component={Link} to="/settings">
          {t("Settings")}
        </MenuItem>
      )}
      <MenuItem onClick={logout} id="menuItem3">
        {t("menu_logout")}
      </MenuItem>
    </Menu>
  );

  const mobileMenuId = "primary-search-account-menu-mobile";
  const renderMobileMenu = (
    <Menu
      anchorEl={mobileMoreAnchorEl}
      anchorOrigin={{ vertical: "top", horizontal: "right" }}
      id={mobileMenuId}
      keepMounted
      transformOrigin={{ vertical: "top", horizontal: "right" }}
      open={isMobileMenuOpen}
      onClose={handleMobileMenuClose}
    >
      <MenuItem onClick={handleProfileMenuOpen}>
        <IconButton
          aria-label="account of current user"
          aria-controls="primary-search-account-menu"
          aria-haspopup="true"
          color="inherit"
        >
          <AccountCircle />
        </IconButton>
        <p>{t("menu_profile")}</p>
      </MenuItem>
    </Menu>
  );

  return (
    <div className={classes.grow} data-testid="app-nav-bar">
      <AppBar position="static">
        <Toolbar>
          <IconButton
            edge="start"
            className={classes.menuButton}
            color="inherit"
            aria-label="open drawer"
          ></IconButton>
          <Typography className={classes.title} variant="h6" noWrap>
            {t("docmanagmentsys")}
          </Typography>
          <div className={classes.grow} />
          <IconButton
            id="IconButton2"
            color="inherit"
            aria-label="open drawer"
            edge="start"
            onClick={() => {
              if (document.getElementsByClassName("help-desk__menu") !== null) {
                document.getElementsByClassName("help-desk__menu")[0].click();
              }
            }}
            className={classes.menuButton}
          >
            <HelpOutlineIcon />
          </IconButton>
          {user_info.userName}
          <div className={classes.sectionDesktop}>
            <IconButton
              id="IconButton3"
              edge="end"
              aria-label="account of current user"
              aria-controls={menuId}
              aria-haspopup="true"
              onClick={handleProfileMenuOpen}
              color="inherit"
            >
              <AccountCircle />
            </IconButton>
          </div>
          <div className={classes.sectionMobile}>
            <IconButton
              id="IconButton4"
              aria-label="show more"
              aria-controls={mobileMenuId}
              aria-haspopup="true"
              onClick={handleMobileMenuOpen}
              color="inherit"
            >
              <MoreIcon />
            </IconButton>
          </div>
        </Toolbar>
      </AppBar>
      {renderMobileMenu}
      {renderMenu}
    </div>
  );
}
