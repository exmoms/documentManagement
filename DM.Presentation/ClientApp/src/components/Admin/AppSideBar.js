import Divider from "@material-ui/core/Divider";
import Drawer from "@material-ui/core/Drawer";
import Hidden from "@material-ui/core/Hidden";
import List from "@material-ui/core/List";
import ListItem from "@material-ui/core/ListItem";
import ListItemIcon from "@material-ui/core/ListItemIcon";
import ListItemText from "@material-ui/core/ListItemText";
import { makeStyles, useTheme } from "@material-ui/core/styles";
import AssignmentIcon from "@material-ui/icons/Assignment";
import DescriptionIcon from "@material-ui/icons/Description";
import FolderOpenIcon from "@material-ui/icons/FolderOpen";
import NoteAddIcon from "@material-ui/icons/NoteAdd";
import SearchIcon from "@material-ui/icons/Search";
import React from "react";
import { useTranslation } from "react-i18next";
import { Link } from "react-router-dom";
// Start: Style properties --------------
const drawerWidth = 260;
const useStyles = makeStyles((theme) => ({
  drawer: {
    [theme.breakpoints.up("sm")]: {
      width: drawerWidth,
      flexShrink: 0,
    },
  },
  toolbar: theme.mixins.toolbar,
  drawerPaper: {
    width: drawerWidth,
    // top: 65,
  },
  content: {
    flexGrow: 1,
    padding: theme.spacing(3),
  },
}));
//End --------------------------------------

// Start: function SideBar ---------------------------------------------------------------
function SideBar(props) {
  const { t } = useTranslation();
  const { container } = props;
  const classes = useStyles();
  const theme = useTheme();

  // list in sidebar ------------
  // Create Forms
  let createFormlist = [
    {
      path: "document-class",
      title: t("label_documentclass"),
      icon: <AssignmentIcon />,
    },
    {
      path: "metadata-model",
      title: t("createdocument_meatdatmodel"),
      icon: <DescriptionIcon />,
    },
  ];
  // -------------------------------------------

  // list in sidebar ------------
  // Full out forms
  let fullOutFormsList = [
    { path: "create-document", title: t("create_doc"), icon: <NoteAddIcon /> },
    {
      path: "browse-documents",
      title: t("browsedocuments"),
      icon: <SearchIcon />,
    },
    {
      path: "document-set",
      title: t("appsidebar_documentset"),
      icon: <FolderOpenIcon />,
    },
  ];
  // -------------------------------------------

  // Start: show the content of sideBar ---------
  const drawer = (
    <>
      <div>
        <div className={classes.toolbar} />
        {(props.user.role === "Admin" || props.user.role === "SuperAdmin") && (
          <React.Fragment>
            <List>
              {createFormlist.map((text, index) => (
                <ListItem
                  button
                  key={text.title}
                  component={Link}
                  to={"/" + text.path}
                >
                  <ListItemIcon>{text.icon}</ListItemIcon>
                  <ListItemText primary={text.title} />
                </ListItem>
              ))}
            </List>
            <Divider />
          </React.Fragment>
        )}
        <List>
          {fullOutFormsList.map((text, index) => (
            <ListItem
              button
              key={text.title}
              component={Link}
              to={"/" + text.path}
            >
              <ListItemIcon>{text.icon}</ListItemIcon>
              <ListItemText primary={text.title} />
            </ListItem>
          ))}
        </List>
      </div>
    </>
  );
  // End: -----------------------------------------------

  return (
    <nav
      className={classes.drawer}
      aria-label="mailbox folders"
      data-testid="app-side-bar"
    >
      {/* The implementation can be swapped with js to avoid SEO duplication of links. */}
      <Hidden smUp implementation="css">
        <Drawer
          id="drawer1"
          container={container}
          variant="temporary"
          anchor={theme.direction === "rtl" ? "right" : "left"}
          open={props.open}
          onClose={props.onClose}
          classes={{
            paper: classes.drawerPaper,
          }}
          ModalProps={{
            keepMounted: true, // Better open performance on mobile.
          }}
        >
          {drawer}
        </Drawer>
      </Hidden>{" "}
      <Hidden xsDown implementation="css">
        <Drawer
          id="drawer2"
          classes={{
            paper: classes.drawerPaper,
          }}
          variant="permanent"
          open
        >
          {drawer}
        </Drawer>
      </Hidden>
    </nav>
  );
}
// End: function SideBar ---------------------------------------------------------------

export default SideBar;
