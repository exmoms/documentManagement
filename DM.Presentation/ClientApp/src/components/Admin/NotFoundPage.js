import React from "react";

export default function () {
  return <div>404 Not found Page!</div>;
}
