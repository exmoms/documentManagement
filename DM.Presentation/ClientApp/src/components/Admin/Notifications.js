import React from "react";
import Typography from "@material-ui/core/Typography";
import { useTranslation } from "react-i18next";
import CloseIcon from "@material-ui/icons/Close";
import { Alert } from "@material-ui/lab";
import IconButton from "@material-ui/core/IconButton";

export function Notification(props) {
  const { t } = useTranslation();
  const { open, error, setOpen, errorMessage } = props;
  return (
    <>
      {open && (
        <Alert
          severity={error ? "error" : "success"}
          action={
            <IconButton
              aria-label="close"
              color="inherit"
              size="small"
              onClick={() => {
                setOpen(false);
              }}
            >
              <CloseIcon fontSize="inherit" />
            </IconButton>
          }
        >
          <Typography variant="subtitle1" align="left" color="primary">
            {error && (
              <ol>
                {errorMessage.map((item, index) => (
                  <li key={index} id={"errorMessageItem"+index}>{item}</li>
                ))}
              </ol>
            )}
            {!error && t("successful_msg")}
          </Typography>
        </Alert>
      )}
    </>
  );
}
