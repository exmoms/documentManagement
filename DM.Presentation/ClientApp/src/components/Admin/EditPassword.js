import React, { useState } from "react";
import Button from "@material-ui/core/Button";
import CssBaseline from "@material-ui/core/CssBaseline";
import TextField from "@material-ui/core/TextField";
import Grid from "@material-ui/core/Grid";
import { makeStyles } from "@material-ui/core/styles";
import Container from "@material-ui/core/Container";
import InputLabel from "@material-ui/core/InputLabel";
// import { postData } from "../../api/PostData";
import { postDataToAPI } from "../../api/PostData";
import { useTranslation } from "react-i18next";
import clsx from "clsx";
import InputAdornment from "@material-ui/core/InputAdornment";
import IconButton from "@material-ui/core/IconButton";
import Visibility from "@material-ui/icons/Visibility";
import VisibilityOff from "@material-ui/icons/VisibilityOff";
import FormControl from "@material-ui/core/FormControl";
import OutlinedInput from "@material-ui/core/OutlinedInput";
import { Notification } from "../Admin/Notifications";

const useStyles = makeStyles((theme) => ({
  paper: {
    marginTop: theme.spacing(8),
    display: "flex",
    flexDirection: "column",
    alignItems: "center",
  },
  form: {
    width: "100%", // Fix IE 11 issue.
    marginTop: theme.spacing(3),
  },
  submit: {
    margin: theme.spacing(3, 0, 2),
  },
  margin: {
    margin: theme.spacing(1),
  },
  textField: {
    width: "100%",
  },
}));

export default function EditPassword() {
  const { t } = useTranslation();
  const classes = useStyles();

  const [password, setPassword] = useState("");
  const [confirmPassword, setConfirmPassword] = useState("");
  const [values, setValues] = useState({
    password: "",
    showPassword: false,
  });

  const regester_file = {
      NewPassword: values.password,
      CurrentPassword: password,
      ConfirmPassword: confirmPassword,
  };

  const [errorMessage, seterrorMessage] = useState([]);
  const [error, seterror] = useState(false);
  const [open, setOpen] = React.useState(false);

  const onchange = (event) => {
    if (event.target.name === "password") {
      setPassword(event.target.value);
    } else if (event.target.name === "confirmPassword") {
      setConfirmPassword(event.target.value);
    }
  };

  const handelSubmit = async (event) => {
    event.preventDefault();
    if (password.length < 8 || password.length > 12) {
      seterror(true);
      seterrorMessage([t("password_charachters_no")]);
      setOpen(true);
    } else if (values.password.length < 8 || confirmPassword.length > 12) {
      seterror(true);
      seterrorMessage([t("password_charachters_no")]);
      setOpen(true);
    } else if (values.password !== confirmPassword) {
      seterror(true);
      seterrorMessage([t("password_mismatch")]);
      setOpen(true);
    } else {
      let response = await postDataToAPI(
        "/api/user/EditPassword",
        regester_file
      );
      if (response.ok) {
        setOpen(true);
      } else {
        let json = await response.json();
        seterror(true);
        seterrorMessage(json.error);
        setOpen(true);
      }
    }
  };

  const handleChange = (prop) => (event) => {
    setValues({ ...values, [prop]: event.target.value });
  };

  const handleClickShowPassword = () => {
    setValues({ ...values, showPassword: !values.showPassword });
  };

  const handleMouseDownPassword = (event) => {
    event.preventDefault();
  };

  return (
    <Container component="main" maxWidth="xs">
      <CssBaseline />
      <div className={classes.paper}>
        <form className={classes.form} onSubmit={handelSubmit} id="submitForm">
          <Grid container spacing={2}>
            <Grid item xs={12}>
              <TextField
                variant="outlined"
                fullWidth
                onChange={onchange}
                name="password"
                label={t("password")}
                type="password"
                id="password"
                autoComplete="current-password"
              />
            </Grid>

            <FormControl
              className={clsx(classes.margin, classes.textField)}
              variant="outlined"
            >
              <InputLabel htmlFor="outlined-adornment-password">
                {t("new_password")}
              </InputLabel>
              <OutlinedInput
                id="outlined-adornment-password"
                type={values.showPassword ? "text" : "password"}
                value={values.password}
                onChange={handleChange("password")}
                endAdornment={
                  <InputAdornment position="end">
                    <IconButton
                    id="iconButton1"
                      aria-label="toggle password visibility"
                      onClick={handleClickShowPassword}
                      onMouseDown={handleMouseDownPassword}
                      edge="end"
                    >
                      {values.showPassword ? <Visibility /> : <VisibilityOff />}
                    </IconButton>
                  </InputAdornment>
                }
                labelWidth={70}
              />
            </FormControl>

            <Grid item xs={12}>
              <TextField
                variant="outlined"
                fullWidth
                onChange={onchange}
                name="confirmPassword"
                label={t("confirmNewPassword_label")}
                type="password"
                id="confirmPassword"
                autoComplete="current-password"
              />
            </Grid>
          </Grid>
          <Button
            type="submit"
            fullWidth
            variant="contained"
            color="primary"
            className={classes.submit}
          >
            {t("buttton_save")}
          </Button>
        </form>
      </div>
      <Notification
      id="notification"
        open={open} // must be a "state" to show notification message [true or false]
        setOpen={setOpen} // this function changes the state "open" value.
        error={error} // if error message put [true] else for success put [false]
        errorMessage={errorMessage} // the recived error message (array of string) from server.
      />
    </Container>
  );
}
