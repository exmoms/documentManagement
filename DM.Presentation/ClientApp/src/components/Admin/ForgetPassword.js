import Avatar from "@material-ui/core/Avatar";
// import Grid from "@material-ui/core/Grid";
import Box from "@material-ui/core/Box";
import Button from "@material-ui/core/Button";
import Container from "@material-ui/core/Container";
import CssBaseline from "@material-ui/core/CssBaseline";
import LinearProgress from "@material-ui/core/LinearProgress";
import { makeStyles } from "@material-ui/core/styles";
import TextField from "@material-ui/core/TextField";
import Typography from "@material-ui/core/Typography";
import LockOutlinedIcon from "@material-ui/icons/LockOutlined";
import React, { useState } from "react";
import { useTranslation } from "react-i18next";
// import FormControlLabel from "@material-ui/core/FormControlLabel";
// import Checkbox from "@material-ui/core/Checkbox";
import { Link, Redirect } from "react-router-dom";
import { postDataToAPI } from "../../api/PostData";
import { useAuth } from "../../context/auth";
import { Notification } from "../Admin/Notifications";

// Start: Style properties --------------
const useStyles = makeStyles((theme) => ({
  paper: {
    marginTop: theme.spacing(8),
    display: "flex",
    flexDirection: "column",
    alignItems: "center",
  },
  avatar: {
    margin: theme.spacing(1),
    backgroundColor: theme.palette.secondary.main,
  },
  main: {
    width: "100%",
    maxWidth: "100%",
  },
  form: {
    width: "100%", // Fix IE 11 issue.
    marginTop: theme.spacing(1),
  },
  submit: {
    margin: theme.spacing(3, 0, 2),
  },
}));
// --------------------------------------

function Copyright() {
  const { t } = useTranslation();
  return (
    <Typography variant="body2" color="textSecondary" align="center">
      {"Copyright Â© "}
      <Link color="inherit" href="http://192.168.0.1/" to="#">
        {t("Company.label")}
      </Link>{" "}
      {new Date().getFullYear()}
      {"."}
    </Typography>
  );
}

export default function ResetPassword() {
  const { t } = useTranslation();
  const classes = useStyles();
  const [loading, setLoading] = useState(false);
  const [isReset, setIsReset] = useState(false);
  const [email, setEmail] = useState("");
  const { authenticated } = useAuth();

  const [open, setOpen] = useState(false);
  const [errorMessage, seterrorMessage] = useState([]);
  const [error, seterror] = useState(false);

  const handelSubmit = async (event) => {
    event.preventDefault();
    if (!loading) {
      setLoading(true);
    }
    let response = await postDataToAPI(
      `/api/user/SendResetPasswordEmail?email=${email}`
    );
    if (response.ok) {
      setIsReset(true);
      setLoading(false);
    } else {
      // let json = await response.json();
      setOpen(true);
      seterror(true);
      seterrorMessage(["Errrrror"] /* json.error */);
      setLoading(false);
    }
  };

  const onChange = (event) => {
    if (event.target.name === "email") {
      setEmail(event.target.value);
    }
  };

  if (authenticated) {
    return <Redirect to="/" />;
  } else {
    if (isReset === false) {
      return (
        <Container component="main" maxWidth="xs" data-testid="loggin">
          <CssBaseline />
          <div className={classes.paper}>
            <Avatar className={classes.avatar}>
              <LockOutlinedIcon />
            </Avatar>
            <Typography component="h1" variant="h5">
              {t("forgetpassword_enterEmail_msg")}
            </Typography>
            <form
              className={classes.form}
              noValidate
              onSubmit={handelSubmit}
              id="submitForm"
            >
              <TextField
                variant="outlined"
                margin="normal"
                required
                fullWidth
                id="email"
                label={t("emailaddress")}
                name="email"
                autoComplete="email"
                autoFocus
                className="emailField"
                onChange={onChange}
              />

              {loading && <LinearProgress color="secondary" />}
              <Button
                type="submit"
                fullWidth
                variant="contained"
                color="primary"
                data-testid="submit"
                className={classes.submit}
              >
                {t("send")}
              </Button>
            </form>
            <Notification
            id="notification"
              open={open} // must be a "state" to show notification message [true or false]
              setOpen={setOpen} // this function changes the state "open" value.
              error={error} // if error message put [true] else for success put [false]
              errorMessage={errorMessage} // the recived error message (array of string) from server.
            />
          </div>
          <Box mt={8}>
            <Copyright />
          </Box>
        </Container>
      );
    } else {
      return (
        <Container
        id="container"
          className={classes.main}
          component="main"
          maxWidth="xs"
          data-testid="loggin"
        >
          <CssBaseline />
          <div className={classes.paper}>
            <Avatar className={classes.avatar}>
              <LockOutlinedIcon />
            </Avatar>
            <Typography component="h1" variant="h5">
              {t("forgetpassword_toContinue_checkMail_msg")}
            </Typography>
          </div>
          <Box mt={8}>
            <Copyright />
          </Box>
        </Container>
      );
    }
  }
}
