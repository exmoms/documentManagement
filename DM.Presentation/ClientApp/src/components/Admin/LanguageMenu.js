import Button from "@material-ui/core/Button";
import Menu from "@material-ui/core/Menu";
import MenuItem from "@material-ui/core/MenuItem";
import TranslateIcon from "@material-ui/icons/Translate";
import React, { useState } from "react";
import { useCookies } from "react-cookie";
import "../../i18n";
import { useTranslation } from "react-i18next";
import { postDataToAPI } from "../../api/PostData";
import { Notification } from "../Admin/Notifications";

const langMap = { en: "English", ar: "العربية", de: "Deutsch" };

export default function LanguageMenu() {
  const [anchorEl, setAnchorEl] = React.useState(null);

  const { i18n } = useTranslation();

  const [cookies] = useCookies(["lang"]);
  var lang = cookies.lang ? cookies.lang.substring(2, 4) : "en";

  localStorage.setItem("i18nextLng", lang);

  const handleClick = (event) => {
    setAnchorEl(event.currentTarget);
  };

  const handleClose = () => {
    setAnchorEl(null);
  };

  const [open, setOpen] = useState(false);
  const [errorMessage, seterrorMessage] = useState([]);
  const [error, seterror] = useState(false);

  const setLanguage = async (event, langCode) => {
    lang = langCode;
    let response = await postDataToAPI("/api/User/ChangeLanguage?lang=" + lang);
    if (response.ok) {
      handleClose();
      i18n.changeLanguage(lang);
      localStorage.setItem("i18nextLng", lang);
    } else {
      // let json = await response.json();
      setOpen(true);
      seterror(true);
      seterrorMessage(["Unable to change language.."] /* json.error */);
    }
  };

  return (
    <React.Fragment>
      <Notification
      id="notification"
        open={open} // must be a "state" to show notification message [true or false]
        setOpen={setOpen} // this function changes the state "open" value.
        error={error} // if error message put [true] else for success put [false]
        errorMessage={errorMessage} // the recived error message (array of string) from server.
      />
      <Button
        aria-controls="language-menu"
        aria-haspopup="true"
        onClick={handleClick}
        variant="contained"
        color="primary"
        startIcon={<TranslateIcon />}
      >
        {langMap[lang]}
      </Button>
      <Menu
        id="language-menu"
        anchorEl={anchorEl}
        keepMounted
        open={Boolean(anchorEl)}
        onClose={handleClose}
      >
        <MenuItem onClick={(event) => setLanguage(event, "en")}>
          English
        </MenuItem>
        <MenuItem onClick={(event) => setLanguage(event, "ar")}>
          العربية
        </MenuItem>
      </Menu>
    </React.Fragment>
  );
}
