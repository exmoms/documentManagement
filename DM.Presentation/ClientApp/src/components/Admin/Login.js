import React, { useState } from "react";
import Avatar from "@material-ui/core/Avatar";
import Button from "@material-ui/core/Button";
import CssBaseline from "@material-ui/core/CssBaseline";
import TextField from "@material-ui/core/TextField";
import { Link } from "react-router-dom";
import Grid from "@material-ui/core/Grid";
import Box from "@material-ui/core/Box";
import LockOutlinedIcon from "@material-ui/icons/LockOutlined";
import Typography from "@material-ui/core/Typography";
import { makeStyles } from "@material-ui/core/styles";
import Container from "@material-ui/core/Container";
import { postDataToAPI } from "../../api/PostData";
import { Notification } from "../Admin/Notifications";
import { useAuth } from "../../context/auth";
import { Redirect } from "react-router-dom";
import LinearProgress from "@material-ui/core/LinearProgress";
import { useTranslation } from "react-i18next";
import "../../i18n";

// Start: Style properties --------------
const useStyles = makeStyles((theme) => ({
  paper: {
    marginTop: theme.spacing(8),
    display: "flex",
    flexDirection: "column",
    alignItems: "center",
  },
  avatar: {
    margin: theme.spacing(1),
    backgroundColor: theme.palette.secondary.main,
  },
  form: {
    width: "100%", // Fix IE 11 issue.
    marginTop: theme.spacing(1),
  },
  submit: {
    margin: theme.spacing(3, 0, 2),
  },
}));
// --------------------------------------

function Copyright() {
  const { t } = useTranslation();
  return (
    <Typography variant="body2" color="textSecondary" align="center">
      {"Copyright Â© "}
      <Link color="inherit" to="" shref="http://192.168.0.1/">
        {t("Company.label")}
      </Link>{" "}
      {new Date().getFullYear()}
      {"."}
    </Typography>
  );
}

export default function SignIn() {
  const { t, i18n } = useTranslation();
  const classes = useStyles();
  const [loading, setLoading] = useState(false);
  const [userName, setUserName] = useState("");
  const [password, setPassword] = useState("");
  const { authenticated, setAuthenticated } = useAuth();

  const [open, setOpen] = useState(false);
  const [errorMessage, seterrorMessage] = useState([]);
  const [error, seterror] = useState(false);

  function getCookie(cname) {
    var name = cname + "=";
    var decodedCookie = decodeURIComponent(document.cookie);
    var ca = decodedCookie.split(";");
    for (var i = 0; i < ca.length; i++) {
      var c = ca[i];
      while (c.charAt(0) === " ") {
        c = c.substring(1);
      }
      if (c.indexOf(name) === 0) {
        return c.substring(name.length, c.length);
      }
    }
    return "";
  }

  const handelSubmit = async (event) => {
    event.preventDefault();
    var login_json = {
      UserName: userName,
      Password: password,
    };
    let response = await postDataToAPI("/api/user/Login", login_json);
    setLoading(true);
    if (response.ok) {
      setLoading(false);
      setAuthenticated(true);

      const langCookie = getCookie("lang");
      var lang = langCookie ? langCookie.substring(2, 4) : "en";
      localStorage.setItem("i18nextLng", lang);
      i18n.changeLanguage(lang);
    } else {
      let json = await response.json();
      setOpen(true);
      seterror(true);
      seterrorMessage(json.error);
      setLoading(false);
    }
  };

  const onChange = (event) => {
    if (event.target.name === "email") {
      setUserName(event.target.value);
    } else if (event.target.name === "password") {
      setPassword(event.target.value);
    }
  };

  if (authenticated) {
    return <Redirect to="/" />;
  } else {
    return (
      <Container component="main" maxWidth="xs" data-testid="loggin">
        <CssBaseline />
        <div className={classes.paper}>
          <Avatar className={classes.avatar}>
            <LockOutlinedIcon />
          </Avatar>
          <Typography component="h1" variant="h5">
            {t("signin")}
          </Typography>
          <form className={classes.form} noValidate onSubmit={handelSubmit}>
            <TextField
              variant="outlined"
              margin="normal"
              required
              fullWidth
              id="email"
              label={t("emailaddress")}
              name="email"
              autoComplete="email"
              autoFocus
              className="emailField"
              onChange={onChange}
            />
            <TextField
              variant="outlined"
              margin="normal"
              required
              fullWidth
              name="password"
              label={t("password")}
              type="password"
              id="password"
              autoComplete="current-password"
              className="passwordField"
              onChange={onChange}
            />
            {loading && <LinearProgress color="secondary" />}
            <Button
              type="submit"
              fullWidth
              variant="contained"
              color="primary"
              data-testid="submit"
              className={classes.submit}
            >
              {"signin"}
            </Button>
            <Grid container>
              <Grid item xs>
                <Link to="/forget-password" variant="body2">
                  {t("forgotpassword")}
                </Link>
              </Grid>
            </Grid>
          </form>
          <Notification
            open={open} // must be a "state" to show notification message [true or false]
            setOpen={setOpen} // this function changes the state "open" value.
            error={error} // if error message put [true] else for success put [false]
            errorMessage={errorMessage} // the recived error message (array of string) from server.
          />
        </div>
        <Box mt={8}>
          <Copyright />
        </Box>
      </Container>
    );
  }
}
