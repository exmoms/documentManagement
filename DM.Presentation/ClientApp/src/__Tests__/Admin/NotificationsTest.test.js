
import React from "react";
import ReactDom from "react-dom";
import { Notification } from "../../components/Admin/Notifications";
import { cleanup } from "@testing-library/react";
import { shallow , mount} from "enzyme";
import toJSON from "enzyme-to-json";
import { Alert } from "@material-ui/lab";
import Typography from "@material-ui/core/Typography";
import IconButton from "@material-ui/core/IconButton";
import { act } from "react-dom/test-utils";
afterEach(cleanup);
describe("Notifications", () => {
   afterEach(() => {
    jest.clearAllMocks();
  });
  test("renders without crashing", () => {
    const div = document.createElement("div");
    let open=true;
    let error=true;
    const setOpen = jest.fn();
    let errorMessage=["error 1" , "error 2"];
    ReactDom.render(<Notification open={open} error={error} setOpen={setOpen} errorMessage={errorMessage} />,div
    );
  });

  test("matches snapshot", () => {
    let open=true;
    let error=true;
    const setOpen = jest.fn();
    let errorMessage=["error 1" , "error 2"];
    const wrapper = shallow(
        <Notification open={open} error={error} setOpen={setOpen} errorMessage={errorMessage}/>
    );
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
   
  test("intializationNotifications", () => {
    let open=true;
    let error=true;
    const setOpen = jest.fn();
    let errorMessage=["error 1" , "error 2"];
    let  wrapper = mount(<Notification open={open} error={error} setOpen={setOpen} errorMessage={errorMessage}/>);
    expect(wrapper.find(Alert).at(0).props().severity).toBe("error");  
    errorMessage.map((item, index) => (
        expect(wrapper.find("#errorMessageItem"+index).text()).toBe(errorMessage[index])
      ));
});
 
test("intializationNotifications when error is false and setOpen is called", () => {
    let open=true;
    let error=false;
    const setOpen = jest.fn();
    let errorMessage=["error 1" , "error 2"];
    let  wrapper = mount(<Notification open={open} error={error} setOpen={setOpen} errorMessage={errorMessage}/>);
    act(() => {
        wrapper
          .find(IconButton)
          .at(0)
          .props()
          .onClick();
      });
      wrapper.update();
    expect(wrapper.find(Alert).at(0).props().severity).toBe("success");  
    errorMessage.map((item, index) => (
        expect(wrapper.find("#errorMessageItem"+index).length).toBe(0)
      ));
      expect(setOpen).toHaveBeenCalledTimes(1);
      expect(wrapper.find(Typography).text()).toBe("successful_msg")
});

test("intializationNotifications when open is false", () => {
    let open=false;
    let error=false;
    const setOpen = jest.fn();
    let errorMessage=["error 1" , "error 2"];
    let  wrapper = mount(<Notification open={open} error={error} setOpen={setOpen} errorMessage={errorMessage}/>);
    expect(wrapper.find(Alert).length).toBe(0);  
});
});