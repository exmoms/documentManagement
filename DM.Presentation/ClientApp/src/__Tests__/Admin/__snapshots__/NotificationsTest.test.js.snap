// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Notifications matches snapshot 1`] = `
<Fragment>
  <WithStyles(ForwardRef(Alert))
    action={
      <ForwardRef(WithStyles)
        aria-label="close"
        color="inherit"
        onClick={[Function]}
        size="small"
      >
        <UNDEFINED
          fontSize="inherit"
        />
      </ForwardRef(WithStyles)>
    }
    severity="error"
  >
    <WithStyles(ForwardRef(Typography))
      align="left"
      color="primary"
      variant="subtitle1"
    >
      <ol>
        <li
          id="errorMessageItem0"
          key="0"
        >
          error 1
        </li>
        <li
          id="errorMessageItem1"
          key="1"
        >
          error 2
        </li>
      </ol>
    </WithStyles(ForwardRef(Typography))>
  </WithStyles(ForwardRef(Alert))>
</Fragment>
`;
